task/backend/scheduler.go

```diff
@@ -359,6 +359,12 @@ func (s *TickScheduler) UpdateTask(authCtx context.Context, task *platform.Task)
 	ts.nextDue = next
 	ts.authCtx = authCtx
 	ts.nextDueMu.Unlock()
+
+	// update the current runners cached task for logs
+	for _, runner := range ts.runners {
+		runner.task = task
```

Comment:
Any potential for a race here? I see we are protected when doing this write by the `schedulerMu`, but I notice the runner itself has its own mutexes for state and it has its hands on the taskSchedulers mutexes as well. Also I notice that the runners also spawn goroutines on themselves.
I guess if the race detector is chill this might not be a problem.

Reply:
Just above we acquire a scheduler wide lock. I will investigate a bit more just incase

Reply:
I looked into it and haven't found any race that could effect us, negatively. If a task is running with the old script and we are logging the run at the same time we may get one last log with the old flux script in it. But this isn't a failure. To fix that we would need to maintain a database transaction through the scheduler.

Reply:
Nice thanks @lyondhill !